src/DSManager.cpp

```diff
@@ -367,14 +381,14 @@ Participant* DSManager::removeParticipant(
 }
 
 void DSManager::addSubscriber(
-        Subscriber* sub)
+        DataReader* sub)
 {
     std::lock_guard<std::recursive_mutex> lock(management_mutex);
-    assert(subscribers[sub->getGuid()] == nullptr);
-    subscribers[sub->getGuid()] = sub;
+    assert(subscribers[sub->guid()] == nullptr);
+    subscribers[sub->guid()] = sub;
```

Comment:
To improve readability and avoid confusion I suggest renaming to datareaders/readers.

```suggestion
    assert(datareaders[reader->guid()] == nullptr);
    datareaders[reader->guid()] = sub;
```